File: src/com/mcblox/parkour/utils/CourseSelect.java
package com.mcblox.parkour.utils;

import java.util.HashMap;
import java.util.Map.Entry;

import org.bukkit.ChatColor;
import org.bukkit.entity.Player;

import com.mcblox.parkour.objects.Course;

public class CourseSelect {

	public static HashMap<Player, Course> selected = new HashMap<Player, Course>();
	
	public static void add(Player player, Course course) {
		if (selected.containsKey(player)) {
			selected.replace(player, course);
		}
		else {
			selected.put(player, course);
		}
	}
	
	public static void remove(Player player) {
		if (selected.containsKey(player)) {
			selected.remove(player);
		}
	}
	
	public static Course get(Player player) {
		return selected.get(player);
	}
	
	public static Player get(Course course) {
		for (Entry<Player, Course> e : selected.entrySet()) {
			if (e.getValue().equals(course)) {
				return e.getKey();
			}
		}
		return null;
	}
	
	public static boolean contains(Player player) {
		return selected.containsKey(player);
	}
	
	public static boolean contains(Course course) {
		return selected.containsValue(course);
	}
	
	public static void noSelectionMessage(Player player) {
		player.sendMessage(ChatColor.RED + "Select a course using /parkour select");
	}
	
}

File: src/org/serversmc/parkour/utils/CourseSelect.kt
package org.serversmc.parkour.utils

import org.bukkit.entity.*
import org.serversmc.parkour.objects.*

object CourseSelect {
	
	private val selected = HashMap<Player, Course>()
	
	fun add(player: Player, course: Course) {
		if (selected.containsKey(player)) {
			selected.replace(player, course)
			return
		}
		selected[player] = course
	}
	
	fun remove(player: Player) {
		if (selected.containsKey(player)) {
			selected.remove(player)
		}
	}
	
	fun contains(player: Player) = selected.contains(player)
	
	fun contains(course: Course) = selected.containsValue(course)
	
	fun get(player: Player) = selected[player]
	
	fun get(course: Course) = ArrayList<Player>().apply { addAll(selected.filterValues { it == course }.keys) }
	
}